src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java

```diff
@@ -119,7 +119,6 @@ private static boolean isEqualsMethod(DetailAST ast) {
         final DetailAST parameters = ast.findFirstToken(TokenTypes.PARAMETERS);
 
         return CheckUtil.isEqualsMethod(ast)
-                && modifiers.findFirstToken(TokenTypes.LITERAL_PUBLIC) != null
```

Comment:
> boolean equals(Object o)

cannot override equals(java.lang.Object) in java.lang.Object attempting to assign weaker access privileges; was public